### PR TITLE
[Improve][connector][kafka] Set default value for partition option

### PR DESCRIPTION
**Pull request by zhilinli123**
…r to specify a default change to the first partition

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by hailin0** [comment]
please ci error

**Comment by zhilinli123** [comment]
> please ci error

@hailin0 Initiate CI. Thank you.

File: seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/sink/KafkaSinkFactory.java [hunk 1]
```diff
@@ -50,7 +50,6 @@ public OptionRule optionRule() {
                         Config.ASSIGN_PARTITIONS,
                         Config.TRANSACTION_PREFIX,
                         Config.SEMANTICS)
-                .exclusive(Config.PARTITION, Config.PARTITION_KEY_FIELDS)
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/sink/KafkaSinkFactory.java:53]: why delete?

Comment (zhilinli123) [reply seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/sink/KafkaSinkFactory.java:53]: This is a mandatory user configuration, and some users don't really need to configure it

@hailin0

Comment (hailin0) [reply seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/sink/KafkaSinkFactory.java:53]: move to optional(...)

**Review by hailin0 — DISMISSED** [review]

File: seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/sink/KafkaSinkWriter.java [hunk 2]
```diff
@@ -174,16 +175,24 @@ private SeaTunnelRowSerializer<byte[], byte[]> getSerializer(
...
+                    "Cannot select both `partiton` and `partition_key_fields`. You can configure only one of them");
+        }
+        if (pluginConfig.get(PARTITION_KEY_FIELDS) != null) {
             return DefaultSeaTunnelRowSerializer.create(
                     topic,
                     getPartitionKeyFields(pluginConfig, seaTunnelRowType),
                     seaTunnelRowType,
                     messageFormat,
                     delimiter);
+        } else if (pluginConfig.get(PARTITION) != null) {
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/sink/KafkaSinkWriter.java]:
```suggestion
if (pluginConfig.get(PARTITION_KEY_FIELDS) != null) {
    return xxxx
}
if (pluginConfig.get(PARTITION) != null) {
    return xxxx
}
return xxxxx;


```
}

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/java/org/apache/seatunnel/e2e/connector/kafka/KafkaIT.java [hunk 3]
```diff
@@ -396,6 +410,28 @@ private Map<String, String> getKafkaConsumerData(String topicName) {
...
+    private List<String> getKafkaConsumerListData(String topicName) {
+        List<String> data = new ArrayList<>();
+        try (KafkaConsumer<String, String> consumer = new KafkaConsumer<>(kafkaConsumerConfig())) {
+            consumer.subscribe(Arrays.asList(topicName));
+            Map<TopicPartition, Long> offsets =
+                    consumer.endOffsets(Arrays.asList(new TopicPartition(topicName, 0)));
+            Long endOffset = offsets.entrySet().iterator().next().getValue();
+            Long lastProcessedOffset = -1L;
+
+            do {
```
Comment (TyrantLucifer) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/java/org/apache/seatunnel/e2e/connector/kafka/KafkaIT.java:422]: `for each` maybe better?

**Review by TyrantLucifer — DISMISSED** [review]

**Review by TyrantLucifer — APPROVED** [review]

**Review by hailin0 — APPROVED** [review]

**Pull request merged**